### PR TITLE
Remove the Firestore Snappy patch

### PR DESCRIPTION
**Pull request by dconeybe**
Remove the logic to patch the `firebase-ios-sdk` dependency to add Snappy support.

This PR will need to be merged when this SDK updates its dependency on the [firebase-ios-sdk](https://github.com/firebase/firebase-ios-sdk) repository since https://github.com/firebase/firebase-ios-sdk/pull/9596 has been merged to add Snappy support directly in the iOS SDK.

The Snappy patching was originally added by #885. This PR effectively reverts that PR now that Snappy support is natively added to the `firebase-ios-sdk` repository.

Googlers see b/227782613 for more details.

### CODE REVIEW

**Comment by jonsimantov** [comment]
Looks like there are some failures, at least on Windows

**Comment by dconeybe** [comment]
> Looks like there are some failures, at least on Windows

I'm not sure what was going on there. I've merged in main to see if that fixes anything. The `desktop.yml` that was used was not consistent with HEAD of the `main` branch.

**Comment by dconeybe** [comment]
@jonsimantov It looks like `ctest` is trying to run the grpc tests, and those tests don't get compiled. This is odd because when I build everything via cmake locally, `ctest` does _not_ try to run the grpc tests. Any ideas why it's trying to run the grpc tests and failing to build them?

**Comment by jonsimantov** [comment]
Hmm, I'm not sure why that would be any different here than in previous occasions. @a-maurice any ideas?

**Comment by dconeybe** [comment]
@jonsimantov @a-maurice Many of the same build errors occur when pointing the C++ SDK at the HEAD of the `master` branch of the firebase-ios-sdk repository: https://github.com/firebase/firebase-cpp-sdk/pull/950. I'm going to move my investigation to that PR until it goes green.

**Comment by dconeybe** [comment]
I'm guessing that the upgrade of grpc in the iOS SDK is the culprit: https://github.com/firebase/firebase-ios-sdk/pull/9488. I think we just need to set the cmake cache variable to disable the "re2" tests (`RE2_BUILD_TESTING`).

**Comment by dconeybe** [comment]
@jonsimantov Can you take a look at this PR? I won't merge it into main but rather into the PR that updates the iOS dependencies.

File: cmake/external/firestore.cmake [hunk 1]
```diff
@@ -28,15 +28,17 @@ function(GetReleasedDep)
     firestore
 
     DOWNLOAD_DIR ${FIREBASE_DOWNLOAD_DIR}
-    URL https://github.com/firebase/firebase-ios-sdk/archive/${version}.tar.gz
+    GIT_REPOSITORY "https://github.com/firebase/firebase-ios-sdk.git"
+    # Pinned HEAD commit as of June 01, 2022 @ 10:51 EDT.
+    GIT_TAG 89c38d1ed908dbc10d5f9f7aded4cf271113773f
```
[the remark this answers is not in the document]
Comment (dconeybe) [reply cmake/external/firestore.cmake:33]: Ahh yes, this particular part of the PR will be reverted. It's just here so I can test again the HEAD of the firebase-ios-sdk.

**Review by jonsimantov — APPROVED** [review]
LGTM, but as noted in the PR, we'll keep this a draft and merge it into the upcoming iOS dependency update.

**Pull request closed without merging**